lib/linter/linter.js

```diff
@@ -314,6 +317,64 @@ function createLintingProblem(options) {
     };
 }
 
+/**
+ * Wraps the value in an Array if it isn't already one.
+ * @template T
+ * @param {T|T[]} value Value to be wrapped.
+ * @returns {Array} The value as an array.
+ */
+function asArray(value) {
+    return Array.isArray(value) ? value : [value];
+}
+
+/**
+ * Pushes a problem to inlineConfigProblems if ruleOptions are redundant.
+ * @param {ConfigData} config Provided config.
+ * @param {Object} loc A line/column location
+ * @param {Array} problems Problems that may be added to.
+ * @param {string} ruleId The rule ID.
+ * @param {Array} ruleOptions The rule options, merged with the config's.
+ * @param {Array} ruleOptionsInline The rule options from the comment.
+ * @param {"error"|"warn"} severity The severity to report.
+ * @returns {void}
+ */
+function addProblemIfSameSeverityAndOptions(config, loc, problems, ruleId, ruleOptions, ruleOptionsInline, severity) {
+    const existingConfigRaw = config.rules?.[ruleId];
+
+    if (!existingConfigRaw) {
+        return;
+    }
+
+    const existingConfig = asArray(existingConfigRaw);
+    const existingSeverity = normalizeSeverityToString(existingConfig[0]);
+    const inlineSeverity = normalizeSeverityToString(ruleOptions[0]);
+    const sameSeverity = existingSeverity === inlineSeverity;
+
+    if (!sameSeverity) {
+        return;
+    }
+
+    let message;
+
+    if (existingConfig.length === 1 && ruleOptions.length === 1) {
+        message = `"${ruleId}" is already configured to ${existingSeverity}.`;
+    } else if (!containsDifferentProperty(ruleOptions.slice(1), existingConfig.slice(1))) {
+        message = ruleOptionsInline.length === 1
+            ? `"${ruleId}" is already configured to ${existingSeverity}.`
+            : `"${ruleId}" is already configured to ${existingSeverity} with the same options.`;
+    }
```

Comment:
```suggestion
    if (existingConfig.length === 1 && ruleOptions.length === 1) {
        message = `Unused inline config ('${ruleId}' is already configured to '${existingSeverity}').`;
    } else if (!containsDifferentProperty(ruleOptions.slice(1), existingConfig.slice(1))) {
        message = ruleOptionsInline.length === 1
            ? `Unused inline config ('${ruleId}' is already configured to '${existingSeverity}').`
            : `Unused inline config ('${ruleId}' is already configured to '${existingSeverity}' with the same options).`;
    }
```

To align the format of these messages with the format of messages for unused disable directives.